zerver/lib/actions.py

```diff
@@ -1030,6 +1031,21 @@ def get_recipient_info(recipient: Recipient,
             'is_muted',
         ).order_by('user_profile_id')
 
+        exists_subquery = subscription_rows.filter(user_profile_id=OuterRef('sender_id'), is_muted=True)
```

Comment:
Let's add a block comment here making clear that in the future we'll likely replace this query with looking at a separate "followed topics" table for greater efficiency.